diff -Nur ansible-2.0.2.0.orig/examples/ansible.cfg ansible-2.0.2.0/examples/ansible.cfg
--- ansible-2.0.2.0.orig/examples/ansible.cfg	2016-04-19 15:55:21.000000000 -0600
+++ ansible-2.0.2.0/examples/ansible.cfg	2016-04-19 17:33:07.274148584 -0600
@@ -222,6 +222,7 @@
 # Example: 
 # control_path = %(directory)s/%%h-%%r
 #control_path = %(directory)s/ansible-ssh-%%h-%%p-%%r
+control_path = %(directory)s/ansible-ssh-%%C
 
 # Enabling pipelining reduces the number of SSH operations required to 
 # execute a module on the remote server. This can result in a significant 
