Index: ansible-1.8.2/lib/ansible/runner/filter_plugins/core.py
===================================================================
--- ansible-1.8.2.orig/lib/ansible/runner/filter_plugins/core.py
+++ ansible-1.8.2/lib/ansible/runner/filter_plugins/core.py
@@ -16,7 +16,12 @@
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
 import base64
-import json
+try:
+    # simplejson is faster and can be more up-to-date (more bugfixes) than the
+    # stdlib json.  In py2.7 stdlib json is based on simplejson 2.0.9
+    import simplejson as json
+except ImportError:
+    import json
 import os.path
 import yaml
 import types
