diff -ur ansible-1.9.3/lib/ansible/modules/core/packaging/os/yum.py ansible-1.9.4/lib/ansible/modules/core/packaging/os/yum.py
--- ansible-1.9.3/lib/ansible/modules/core/packaging/os/yum.py	2015-09-03 16:03:12.000000000 -0700
+++ ansible-1.9.4/lib/ansible/modules/core/packaging/os/yum.py	2015-09-04 10:03:32.000000000 -0700
@@ -733,7 +733,7 @@
     rc, out, err = module.run_command(yum_basecmd + ['check-update'])
     if rc == 0 and update_all:
         res['results'].append('Nothing to do here, all packages are up to date')
-        return res
+        module.exit_json(**res)
     elif rc == 100:
         available_updates = out.split('\n')
         # build update dictionary
@@ -813,7 +813,7 @@
         if len(will_update) > 0 or len(pkgs['install']) > 0:
             res['changed'] = True
 
-        return res
+        module.exit_json(**res)
 
     # run commands
     if cmd:     # update all
