diff -Nur ansible-1.9.3.orig/lib/ansible/module_utils/facts.py ansible-1.9.3/lib/ansible/module_utils/facts.py
--- ansible-1.9.3.orig/lib/ansible/module_utils/facts.py	2015-09-03 17:03:08.000000000 -0600
+++ ansible-1.9.3/lib/ansible/module_utils/facts.py	2015-09-03 20:19:33.256457164 -0600
@@ -115,6 +115,7 @@
     # package manager, put the preferred one last.  If there is an
     # ansible module, use that as the value for the 'name' key.
     PKG_MGRS = [ { 'path' : '/usr/bin/yum',         'name' : 'yum' },
+                 { 'path' : '/usr/bin/dnf',         'name' : 'dnf' },
                  { 'path' : '/usr/bin/apt-get',     'name' : 'apt' },
                  { 'path' : '/usr/bin/zypper',      'name' : 'zypper' },
                  { 'path' : '/usr/sbin/urpmi',      'name' : 'urpmi' },
